File: fabric/src/main/java/wile/redstonepen/gametest/InventoriesGameTests.java
package wile.redstonepen.gametest;

import net.minecraft.gametest.framework.GameTest;
import net.minecraft.gametest.framework.GameTestHelper;
import wile.redstonepen.gametestcommon.InventoriesTests;

public class InventoriesGameTests
{
  private static final String RELAY_TEMPLATE = "redstonepen:relay_activates_from_redstone";

  public InventoriesGameTests() {}

  @GameTest(template = RELAY_TEMPLATE, timeoutTicks = 5)
  public static void inventoryRangeInsertBetweenMatchedSlots(GameTestHelper helper)
  { InventoriesTests.inventoryRangeInsertBetweenMatchedSlots(helper); }

  @GameTest(template = RELAY_TEMPLATE, timeoutTicks = 5)
  public static void inventoryRangeInsertBeforeAfterMatchedSlot(GameTestHelper helper)
  { InventoriesTests.inventoryRangeInsertBeforeAfterMatchedSlot(helper); }

  @GameTest(template = RELAY_TEMPLATE, timeoutTicks = 5)
  public static void inventoryRangeInsertSimulateWithPartialStack(GameTestHelper helper)
  { InventoriesTests.inventoryRangeInsertSimulateWithPartialStack(helper); }

  @GameTest(template = RELAY_TEMPLATE, timeoutTicks = 5)
  public static void inventoryRangeInsertSimulateNoRoomReturnsInput(GameTestHelper helper)
  { InventoriesTests.inventoryRangeInsertSimulateNoRoomReturnsInput(helper); }

  @GameTest(template = RELAY_TEMPLATE, timeoutTicks = 5)
  public static void inventoryRangeExtractMultipleMatchingSlots(GameTestHelper helper)
  { InventoriesTests.inventoryRangeExtractMultipleMatchingSlots(helper); }

  @GameTest(template = RELAY_TEMPLATE, timeoutTicks = 5)
  public static void inventoryRangeExtractWithSimulateTrue(GameTestHelper helper)
  { InventoriesTests.inventoryRangeExtractWithSimulateTrue(helper); }

  @GameTest(template = RELAY_TEMPLATE, timeoutTicks = 5)
  public static void inventoryRangeInsertFillupOnlyStopsAtOnlyFillup(GameTestHelper helper)
  { InventoriesTests.inventoryRangeInsertFillupOnlyStopsAtOnlyFillup(helper); }
}

File: fabric/src/main/kotlin/wile/redstonepen/gametest/InventoriesGameTests.kt
package wile.redstonepen.gametest

import net.minecraft.gametest.framework.GameTest
import net.minecraft.gametest.framework.GameTestHelper
import wile.redstonepen.gametestcommon.InventoriesTests

@Suppress("UtilityClassWithPublicConstructor")
class InventoriesGameTests {
    companion object {
        private const val RELAY_TEMPLATE = "redstonepen:relay_activates_from_redstone"

        @JvmStatic
        @GameTest(template = RELAY_TEMPLATE, timeoutTicks = 5)
        fun inventoryRangeInsertBetweenMatchedSlots(helper: GameTestHelper) =
            InventoriesTests.inventoryRangeInsertBetweenMatchedSlots(helper)

        @JvmStatic
        @GameTest(template = RELAY_TEMPLATE, timeoutTicks = 5)
        fun inventoryRangeInsertBeforeAfterMatchedSlot(helper: GameTestHelper) =
            InventoriesTests.inventoryRangeInsertBeforeAfterMatchedSlot(helper)

        @JvmStatic
        @GameTest(template = RELAY_TEMPLATE, timeoutTicks = 5)
        fun inventoryRangeInsertSimulateWithPartialStack(helper: GameTestHelper) =
            InventoriesTests.inventoryRangeInsertSimulateWithPartialStack(helper)

        @JvmStatic
        @GameTest(template = RELAY_TEMPLATE, timeoutTicks = 5)
        fun inventoryRangeInsertSimulateNoRoomReturnsInput(helper: GameTestHelper) =
            InventoriesTests.inventoryRangeInsertSimulateNoRoomReturnsInput(helper)

        @JvmStatic
        @GameTest(template = RELAY_TEMPLATE, timeoutTicks = 5)
        fun inventoryRangeExtractMultipleMatchingSlots(helper: GameTestHelper) =
            InventoriesTests.inventoryRangeExtractMultipleMatchingSlots(helper)

        @JvmStatic
        @GameTest(template = RELAY_TEMPLATE, timeoutTicks = 5)
        fun inventoryRangeExtractWithSimulateTrue(helper: GameTestHelper) =
            InventoriesTests.inventoryRangeExtractWithSimulateTrue(helper)

        @JvmStatic
        @GameTest(template = RELAY_TEMPLATE, timeoutTicks = 5)
        fun inventoryRangeInsertFillupOnlyStopsAtOnlyFillup(helper: GameTestHelper) =
            InventoriesTests.inventoryRangeInsertFillupOnlyStopsAtOnlyFillup(helper)
    }
}
